packages/next/client/index.tsx

```diff
@@ -397,8 +397,13 @@ function renderError(renderErrorProps: RenderErrorProps): Promise<any> {
           AppTree,
         },
       }
+      const isInitialRenderError = !!renderErrorProps.props?.err
+      const hasGetInitialProps = Boolean(
+        (appCtx.Component as any).getInitialProps
```

Comment:
do we need to check `App.getInitialProps` as well?

It doesn't seem necessary to keep the https://github.com/vercel/next.js/pull/21240 use-case working.